### PR TITLE
[INLONG-3112][Manager] Support metadata field in manager and manager client

### LINKED ISSUE #3112: [Feature][Manager] Support metadata in manager and manager client
### Description

Support metadata in manager and manager client when data integration, such as  mysql binlog medata, canal json and so on

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Title Name: [INLONG-3112][Manager] Support metadata in manager and manager client

Fixes #3112 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Review by EMsnap — APPROVED** [review]

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/SinkField.java [hunk 1]
```diff
@@ -33,10 +33,14 @@
     @ApiModelProperty("Source field type")
     private String sourceFieldType;
 
+    @ApiModelProperty("Is source builtin, 0: no, 1: yes")
+    private Integer isSourceBuiltin = 0;
+
     public SinkField(int index, FieldType fieldType, String fieldName, String fieldComment,
-            String fieldValue, String sourceFieldName, String sourceFieldType) {
+            String fieldValue, String sourceFieldName, String sourceFieldType, Integer isSourceBuiltin) {
```
Comment (kipshi) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/SinkField.java:40]: 'isSourceBuiltin' is difficult to understand ,  use metaField instead?

Comment (yunqingmoswu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/SinkField.java]: done

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java [hunk 2]
```diff
@@ -152,7 +152,8 @@ private static StreamSink parseClickHouseSink(ClickHouseSinkResponse sinkRespons
                 sinkFieldResponse.getFieldName(),
                 sinkFieldResponse.getFieldComment(),
                 null, sinkFieldResponse.getSourceFieldName(),
-                sinkFieldResponse.getSourceFieldType())).collect(Collectors.toList());
+                sinkFieldResponse.getSourceFieldType(),
+                sinkFieldResponse.getIsSourceBuiltin())).collect(Collectors.toList());
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:156]: Replace with isSourceMetaField pls.

File: inlong-manager/manager-web/sql/apache_inlong_manager.sql [hunk 3]
```diff
@@ -16,13 +16,16 @@
  */
 
 SET NAMES utf8mb4;
-SET FOREIGN_KEY_CHECKS = 0;
+SET
```
Comment (healchow) [inline inlong-manager/manager-web/sql/apache_inlong_manager.sql:19]: Please not change the code style of SQL file.

Comment (healchow) [reply inlong-manager/manager-web/sql/apache_inlong_manager.sql:19]: In addition, the SQL file under the manager-test module also needs to be modified.

Comment (yunqingmoswu) [reply inlong-manager/manager-web/sql/apache_inlong_manager.sql:19]: done

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java (hunk 2 above)
Comment (yunqingmoswu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java]: done

File: inlong-manager/manager-dao/src/main/resources/mappers/StreamSinkFieldEntityMapper.xml [hunk 4]
```diff
@@ -31,24 +32,28 @@
         <result column="field_comment" jdbcType="VARCHAR" property="fieldComment"/>
         <result column="source_field_name" jdbcType="VARCHAR" property="sourceFieldName"/>
         <result column="source_field_type" jdbcType="VARCHAR" property="sourceFieldType"/>
+        <result column="is_source_meta_field" jdbcType="INTEGER" property="isSourceMetaField"/>
         <result column="rank_num" jdbcType="SMALLINT" property="rankNum"/>
         <result column="is_deleted" jdbcType="INTEGER" property="isDeleted"/>
     </resultMap>
     <sql id="Base_Column_List">
-        id, sink_id, field_name, field_type, field_comment,
+        id
```
Comment (healchow) [inline inlong-manager/manager-dao/src/main/resources/mappers/StreamSinkFieldEntityMapper.xml:40]: Please add `is_source_meta_field` in here, and not change the code style.

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**